barretenberg/cpp/src/barretenberg/goblin/goblin.hpp

```diff
@@ -109,7 +112,7 @@ class Goblin {
         auto ultra_proof = prover.construct_proof();
 
         // Construct and store the merge proof to be recursively verified on the next call to accumulate
-        auto merge_prover = composer.create_merge_prover(op_queue);
+        MergeProver merge_prover{ op_queue };
```

Comment:
Is the plan to eventually do this for all Provers/Verifiers or is there sth specific to the Merge ones that motivated this change?

Reply:
For the Merge classes it didn't make any sense for the Composer to be involved, this was just leftover from when the merge protocol was baked into the main Ultra protocol. More generally, it makes sense for classes to be constructed directly via their constructors when possible. The only argument for having methods like create_prover/verifier on the Composer is that the composer is doing some behind the scenes work that's required for the construction of both of these classes. I suspect that we could do what I've done here for most of the other classes currently constructed by the composer and I think we probably should.